webui/ocw/lib/db.py

```diff
@@ -54,47 +55,53 @@ def __update_run():
     from . import EC2db
     from . import azure
     from . import gce
+    global update_error
 
     '''
     Each update is using Instance.active to mark the model is still availalbe on CSP.
     Instance.state is used to reflect the "local" state, e.g. if someone triggered a delete, the
     state will moved to DELETING. If the instance is gone from CSP, the state will set to DELETED.
     '''
+    try:
+        instances = Azure().list_resource_groups()
+        logger.info('Got {} resources groups from Azure'.format(len(instances)))
+        azure.sync_instances_db(instances)
 
-    instances = Azure().list_resource_groups()
-    logger.info('Got {} resources groups from Azure'.format(len(instances)))
-    azure.sync_instances_db(instances)
+        for region in EC2().list_regions():
+            instances = EC2().list_instances(region=region)
+            logger.info('Got {} instances from EC2 in region {}'.format(len(instances), region))
+            EC2db.sync_instances_db(region, instances)
 
-    for region in EC2().list_regions():
-        instances = EC2().list_instances(region=region)
-        logger.info('Got {} instances from EC2 in region {}'.format(len(instances), region))
-        EC2db.sync_instances_db(region, instances)
-
-    instances = GCE().list_all_instances()
-    logger.info('Got {} instances from GCE'.format(len(instances)))
-    gce.sync_instances_db(instances)
+        instances = GCE().list_all_instances()
+        logger.info('Got {} instances from GCE'.format(len(instances)))
+        gce.sync_instances_db(instances)
+    except Exception as e:
+        logger.exception('Unexpected error on update instances')
```

Comment:
why you not writing `e` content into a log ?

Reply:
logger.exception() does it. https://docs.python.org/3/library/logging.html#logging.Logger.exception

Reply:
ah ok